### PR TITLE
Refactored WS config

### PR DESCRIPTION
**Pull request by jroper**
* Moved to standard Typesafe config reference.conf structure
* Namespaced config to play.ws, with deprecation warnings for old config locations
* Eliminated traits of config objects
* Switched to values in config objects rather than Option, except where the default should be decided by the system and the system provides no easy way to get it
* Kept most config the same, except ssl.debug, which moved to many booleans rather than a list of strings
* Added scaladocs everywhere

### CODE REVIEW

**Comment by jroper** [comment]
Not to be merged until #4093 is merged.

**Comment by jroper** [comment]
This is ready to be merged.

File: framework/src/play-ws/src/main/resources/reference.conf [hunk 1]
```diff
@@ -3,4 +3,227 @@ play {
     enabled += "play.api.libs.ws.ning.NingWSModule"
     enabled += "play.api.libs.openid.OpenIDModule"
   }
+  # Configuratino for Play ws
```
Comment (richdougherty) [inline framework/src/play-ws/src/main/resources/reference.conf]: Typo

File: framework/src/play-ws/src/main/scala/play/api/libs/ws/ssl/Config.scala [hunk 2]
```diff
@@ -366,77 +317,57 @@ class DefaultSSLConfigParser(c: Configuration, classLoader: ClassLoader) {
...
-  def parseKeyManager(config: Configuration): KeyManagerConfig = {
-
-    val algorithm = config.getString("algorithm")
+  def parseKeyManager(config: PlayConfig): KeyManagerConfig = {
 
-    val password = config.getString("password")
+    val algorithm = config.getOptional[String]("algorithm") match {
+      case None => KeyManagerFactory.getDefaultAlgorithm
+      case Some(other) => other
+    }
```
Comment (richdougherty) [inline framework/src/play-ws/src/main/scala/play/api/libs/ws/ssl/Config.scala]: Use `getOrElse` instead?

File: framework/src/play-ws/src/main/scala/play/api/libs/ws/ssl/Config.scala [hunk 3]
```diff
@@ -366,77 +317,57 @@ class DefaultSSLConfigParser(c: Configuration, classLoader: ClassLoader) {
...
-          parseTrustStoreInfo(storeConfig)
-        }
-    }.getOrElse(Nil)
-
-    DefaultTrustManagerConfig(algorithm, trustStoreInfos)
+  def parseTrustManager(config: PlayConfig): TrustManagerConfig = {
+    val algorithm = config.getOptional[String]("algorithm") match {
+      case None => TrustManagerFactory.getDefaultAlgorithm
+      case Some(other) => other
+    }
```
Comment (richdougherty) [inline framework/src/play-ws/src/main/scala/play/api/libs/ws/ssl/Config.scala]: `getOrElse`?

**Comment by richdougherty** [comment]
> * Eliminated traits of config objects
> * Switched to values in config objects rather than Option, except where the default should be decided by the system and the system provides no easy way to get it

Both of these changes tidied up the code a lot. Very nice.

**Pull request merged**